data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java

```diff
@@ -119,12 +127,16 @@ private IndexConfiguration(final Builder builder) {
         this.routingField = builder.routingField;
 
         String documentIdField = builder.documentIdField;
+        String documentId = builder.documentId;
         if (indexType.equals(IndexType.TRACE_ANALYTICS_RAW)) {
             documentIdField = "spanId";
+            documentId = "spanId";
         } else if (indexType.equals(IndexType.TRACE_ANALYTICS_SERVICE_MAP)) {
             documentIdField = "hashId";
+            documentId = "hashId";
```

Comment:
Again, shouldn't we remove `documentIdField = "hashId";`?

Reply:
That would be a breaking change wouldn't it?

Reply:
Oh I see I'll add a check to keep the null one null